EosPayload/drivers/efield_driver.py

```diff
@@ -0,0 +1,87 @@
+import logging
+import traceback
+
+
+from EosLib.format.formats.e_field import EField
+from EosLib.format import Type
+from EosLib.packet import Packet
+from EosLib.packet.data_header import DataHeader
+from EosLib.packet.definitions import Priority
+
+from EosPayload.lib.base_drivers.driver_base import DriverBase
+from EosPayload.lib.mqtt import Topic
+
+try:
+    import Adafruit_BBIO.ADC as ADC
+except ModuleNotFoundError:
+    print("ADC not installed correctly")
+    pass
+
+
+class ElectricFieldSensor(DriverBase):
+    def __init__(self, output_directory: str, config: dict):
+        super().__init__(output_directory, config)
+        self.pin_1 = "P9_36"
+        self.pin_2 = "P9_38"
+        self.pin_3 = "P9_40"
+
+    def setup(self) -> None:
+        super().setup()
+        try:
+            ADC
+        except NameError:
+            raise Exception("failed to import ADC library")
+
+        self.register_thread('device-read', self.device_read)
+
+        ADC.setup(self.pin_1)
+        ADC.setup(self.pin_2)
+        ADC.setup(self.pin_3)
+
+    def device_read(self, logger: logging.Logger) -> None:
+        adc_pins = [self.pin_1, self.pin_2, self.pin_3]
+        voltages = [None, None, None]
+        while True:
+            try:
+                # Read the voltage from the ADC pin
+                voltages = [ADC.read(pin) for pin in adc_pins]
```

Comment:
you should restructure this into a loop so you can read the pins independently.  that way if an exception is raised on reading pin1, we can still try to read pins 2 and 3